boards/arm/npcx7m6fb_evb/npcx7m6fb_evb.dts

```diff
@@ -76,8 +76,10 @@
 };
 
 &pwm6 {
-	drive-open-drain;
 	status = "okay";
+	pinctrl-0 = <&pwm6_gpc0>;
+	pinctrl-names = "default";
+	drive-open-drain;
```

Comment:
this property should be part of the `pwm6_gpc0` node

Reply:
Done. Setting bit 7 in [PWMCTLEX](https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/pwm/pwm_npcx.c#L84) selects the type of the PWM_n output buffer to push-pull or open-drain. I don't know it's a good idea to move it to pinctrl_npcx.c. Hence, I still keep it in the pwm driver.

Reply:
Please review the latest PR.